packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmMapInternal.kt

```diff
@@ -264,7 +273,7 @@ internal class RealmObjectMapOperator<K, V>(
                 )
                 Pair(previousObject, it.second)
             } as Pair<V?, Boolean>
-        }
+        }.also { modCount += 1 }
```

Comment:
Why not `modCount++`?

Reply:
The `++` operator doesn't modify the instance on which it is executed, it returns the modified value 🤯 see https://kotlinlang.org/docs/operator-overloading.html#increments-and-decrements